src/utils/cellSetOperations.js

```diff
@@ -22,11 +22,7 @@ const union = (listOfSets, properties) => {
 
   const sets = listOfSets.map((key) => properties[key]?.cellIds || []);
   const unionSet = new Set(
-    [].concat(
-      ...sets.map(
-        (set) => [...set],
-      ),
-    ),
+    sets.flatMap(set => [...set]),
```

Comment:
Nice, do you know if we could make it even simpler and use `flat()`?

Reply:
No it won't work. Sets is an array of Set objects, then sets.flat() won't work because Set objects aren't arrays. In this context, sets.flatMap(set => [...set]) is appropriate to use because it first converts the Set to an array and then flattens it.

Reply:
ah ok